services/scabbard/libscabbard/src/service/factory/factory_static_configuration.rs

```diff
@@ -457,95 +432,65 @@ impl ServiceFactory for ScabbardFactory {
-            ScabbardFactoryStorageConfig::Lmdb { db_dir, db_size } => {
-                let receipt_db_dir_path = Path::new(&db_dir);
-
-                let receipt_db_path =
-                    compute_db_path(&service_id, circuit_id, receipt_db_dir_path, "receipts")?;
-
-                let file: String = receipt_db_path
-                    .with_extension("lmdb")
-                    .components()
-                    .rev()
-                    .next()
-                    .map(|p| {
-                        let p: &std::path::Path = p.as_ref();
-                        p.to_str().map(|s| s.to_string())
-                    })
-                    .flatten()
-                    .ok_or_else(|| {
-                        FactoryCreateError::Internal(
-                            "File name produced was not valid UTF-8".into(),
-                        )
-                    })?;
-
-                let purge_db_path = receipt_db_path;
-                (
-                    Arc::new(RwLock::new(
-                        LmdbReceiptStore::new(
-                            receipt_db_dir_path,
-                            &[file.clone()],
-                            file,
-                            Some(*db_size),
-                        )
-                        .map_err(|e| FactoryCreateError::Internal(e.to_string()))?,
-                    )),
-                    Box::new(move || {
-                        purge_paths(&purge_db_path)?;
-                        Ok(())
-                    })
-                        as Box<dyn Fn() -> Result<(), InternalError> + Sync + Send>,
-                )
-            }
-            #[cfg(all(feature = "receipt-store", feature = "postgres"))]
-            ScabbardFactoryStorageConfig::Postgres { pool } => (
-                Arc::new(RwLock::new(DieselReceiptStore::new(
-                    pool.clone(),
-                    Some(format!("{}::{}", circuit_id, service_id)),
-                ))),
-                Box::new(|| Ok(())) as Box<dyn Fn() -> Result<(), InternalError> + Sync + Send>,
-            ),
-            #[cfg(all(feature = "receipt-store", feature = "sqlite"))]
-            ScabbardFactoryStorageConfig::Sqlite { pool } => (
-                Arc::new(RwLock::new(DieselReceiptStore::new(
-                    pool.clone(),
-                    Some(format!("{}::{}", circuit_id, service_id)),
-                ))),
-                Box::new(|| Ok(())) as Box<dyn Fn() -> Result<(), InternalError> + Sync + Send>,
-            ),
-            #[cfg(all(
-                feature = "receipt-store",
-                not(any(feature = "postgres", feature = "sqlite"))
-            ))]
-            _ => {
-                return Err(FactoryCreateError::Internal(
-                    "receipt-store was enabled, but without any databases".into(),
-                ))
-            }
-        };
+        let (receipt_store, receipt_purge): (Arc<RwLock<dyn ReceiptStore>>, _) =
```

Comment:
This also looks unrelated, removing the use of ScabbardReceiptStore. Should at minimum be a separate commit.